### ISSUE TITLE
This does not support the Minecraft education edition level.dat files.

### ISSUE THREAD

**Issue by DudleyFox** [vberlier/nbtlib#3]
This does not support the Minecraft Education Edition level.dat files out of the box. There are two main reasons:

1.  It doesn't understand the eight byte header at the start of the file. The first four bytes look like a version or marker field (in my file it is an 8). The next four bytes are the length of the file in bytes not counting the eight byte header.
1. The numbers in the file are all stored in little endian format.

I was able to read my particular file by chopping out the first 8 bytes of the file and changing this block of code in tag.py:
```
# Struct formats used to pack and unpack numeric values

BYTE = struct.Struct('<b')
SHORT = struct.Struct('<h')
USHORT = struct.Struct('<H')
INT = struct.Struct('<i')
LONG = struct.Struct('<q')
FLOAT = struct.Struct('<f')
DOUBLE = struct.Struct('<d')

**Comment by vberlier** [comment]
`nbtlib` currently uses big-endian byte ordering, which is the format used by the original version of Minecraft. I know that versions of Minecraft based on the pocket edition use little-endian ordering, so making it possible to choose the endianness when reading and writing nbt data is on my todo list.

However, `nbtlib` is not a library for working with Minecraft files specifically. It's an implementation of the official nbt specification. Nbt data is used in a lot of different ways, and it's often integrated within other file formats. This library is supposed to let you read sections of files that contain spec-conforming nbt data. It won't work out of the box on files that do not contain raw nbt data, such as the Minecraft Education `level.dat` file since it has a header that you'll need to parse yourself before getting to the section that contains actual nbt data.

**Comment by vberlier** [comment]
@DudleyFox I just published version 1.2.0, which now supports little-endian nbt data. I added a `byteorder` argument to most functions and methods. It defaults to `'big'` for backward compatibility but you can set it to `'little'` to use the little-endian format.

```python
parsed_compound = Compound.parse(some_file, byteorder='little')
```

I also added an example in the [examples](https://github.com/vberlier/nbtlib/tree/master/examples) folder that showcases how you could use `nbtlib` to read and edit level.dat files from Bedrock-based Minecraft editions. The `level_dat_bedrock.py` example defines a `BedrockLevelFile` class that takes care of parsing the header and the content of the file using the little-endian format.

```python
# This is how you would edit a level.dat file

with BedrockLevelFile.load('level.dat') as level_data:
    level_data['LevelName'] = 'Something'
```

For the sake of convenience, I attached a schema that matches the structure of the level.dat file to the `BedrockLevelFile` class, but that's completely optional.
